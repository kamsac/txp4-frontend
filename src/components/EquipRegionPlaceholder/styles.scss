@import '../../variables';

.EquipRegionPlaceholder {
  display: grid;
  grid-gap: $grid-gap-small/2;
  grid-template-areas:
    "InventoryItem"
    "EquippedItemDetails";
  grid-template-columns: 1fr;
  grid-template-rows: 100px 1fr;
}

.EquipRegionPlaceholder__Item {
  background-color: $gray-light;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  border: $border-width solid $black;
  border-radius: $border-radius;
}

.EquipRegionPlaceholder__Details {
  border: $border-width solid $black;
  border-radius: $border-radius;
  grid-area: EquippedItemDetails;
}

.EquipRegionPlaceholder__Title {
  font-size: $font-size-header;
}
