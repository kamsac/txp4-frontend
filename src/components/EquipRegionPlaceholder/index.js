import React from 'react';
import './styles.scss';

export default class EquipRegionPlaceholder extends React.Component{
  constructor(props) {
    super(props);
  }

  render() {
    const imageClassName = `InventoryItem--${this.props.equipRegion}`;
    const itemClassName = `EquipRegionPlaceholder__Item ${imageClassName}`;

    return (
      <div className="EquipRegionPlaceholder">
        <div className={itemClassName} />
        <div className="EquipRegionPlaceholder__Details">
          <h1 className="EquipRegionPlaceholder__Title">{this.props.equipRegion}</h1>
          <p>No custom {this.props.equipRegion} is equipped.</p>
        </div>
      </div>
    )
  };
}
