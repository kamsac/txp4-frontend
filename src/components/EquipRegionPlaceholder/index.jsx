import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

export default class EquipRegionPlaceholder extends React.Component {
  render() {
    const imageClassName = `InventoryItem--${this.props.equipRegion}`;
    const itemClassName = `EquipRegionPlaceholder__Item ${imageClassName}`;

    return (
      <div className="EquipRegionPlaceholder">
        <div className={itemClassName} />
        <div className="EquipRegionPlaceholder__Details">
          <h1 className="EquipRegionPlaceholder__Title">{this.props.equipRegion}</h1>
          <p className="EquipRegionPlaceholder__Description">No custom {this.props.equipRegion} is equipped.</p>
        </div>
      </div>
    );
  }
}

EquipRegionPlaceholder.propTypes = {
  equipRegion: PropTypes.string.isRequired,
};
