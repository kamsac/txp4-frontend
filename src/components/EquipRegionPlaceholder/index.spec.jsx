import React from 'react';
import { shallow } from 'enzyme';
import EquipRegionPlaceholder from './index';

test('<EquipRegionPlaceholder>', () => {
  const component = shallow(<EquipRegionPlaceholder equipRegion="tire" />);
  expect(component).toMatchSnapshot();
});
