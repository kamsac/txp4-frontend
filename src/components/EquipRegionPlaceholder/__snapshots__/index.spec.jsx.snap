// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EquipRegionPlaceholder> 1`] = `
<div
  className="EquipRegionPlaceholder"
>
  <div
    className="EquipRegionPlaceholder__Item InventoryItem--tire"
  />
  <div
    className="EquipRegionPlaceholder__Details"
  >
    <h1
      className="EquipRegionPlaceholder__Title"
    >
      tire
    </h1>
    <p>
      No custom 
      tire
       is equipped.
    </p>
  </div>
</div>
`;
